From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 18 May 2015 09:43:07 +0200
Subject: Revert "xen/xenbus: Update xenbus event channel on resume"
Patch-mainline: never, only a temporary band-aid

This reverts commit 2a4de06fb56014a179a0fd5b788cd295c3e6fcbf as it
collides badly with xen patches. Waiting for master to keep up with
the changes.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/xen/xenbus/xenbus_probe.c | 29 -----------------------------
 1 file changed, 29 deletions(-)

diff --git a/drivers/xen/xenbus/xenbus_probe.c b/drivers/xen/xenbus/xenbus_probe.c
index 5390a674b5e3..564b31584860 100644
--- a/drivers/xen/xenbus/xenbus_probe.c
+++ b/drivers/xen/xenbus/xenbus_probe.c
@@ -57,7 +57,6 @@
 #include <xen/xen.h>
 #include <xen/xenbus.h>
 #include <xen/events.h>
-#include <xen/xen-ops.h>
 #include <xen/page.h>
 
 #include <xen/hvm.h>
@@ -736,30 +735,6 @@ static int __init xenstored_local_init(void)
 	return err;
 }
 
-static int xenbus_resume_cb(struct notifier_block *nb,
-			    unsigned long action, void *data)
-{
-	int err = 0;
-
-	if (xen_hvm_domain()) {
-		uint64_t v;
-
-		err = hvm_get_parameter(HVM_PARAM_STORE_EVTCHN, &v);
-		if (!err && v)
-			xen_store_evtchn = v;
-		else
-			pr_warn("Cannot update xenstore event channel: %d\n",
-				err);
-	} else
-		xen_store_evtchn = xen_start_info->store_evtchn;
-
-	return err;
-}
-
-static struct notifier_block xenbus_resume_nb = {
-	.notifier_call = xenbus_resume_cb,
-};
-
 static int __init xenbus_init(void)
 {
 	int err = 0;
@@ -818,10 +793,6 @@ static int __init xenbus_init(void)
 		goto out_error;
 	}
 
-	if ((xen_store_domain_type != XS_LOCAL) &&
-	    (xen_store_domain_type != XS_UNKNOWN))
-		xen_resume_notifier_register(&xenbus_resume_nb);
-
 #ifdef CONFIG_XEN_COMPAT_XENFS
 	/*
 	 * Create xenfs mountpoint in /proc for compatibility with
-- 
2.3.7

